launcher/ui/dialogs/ResourceDownloadDialog.cpp

```diff
@@ -117,18 +123,63 @@ void ResourceDownloadDialog::connectButtons()
     connect(HelpButton, &QPushButton::clicked, m_container, &PageContainer::help);
 }
 
+static ModPlatform::ProviderCapabilities ProviderCaps;
+
+QStringList getReqiredBy(QList<ResourceDownloadDialog::DownloadTaskPtr> tasks, QVariant addonId)
+{
+    auto req = QStringList();
+    for (auto& task : tasks) {
+        auto deps = task->getVersion().dependencies;
+        if (auto dep = std::find_if(deps.begin(), deps.end(),
+                                    [addonId](const ModPlatform::Dependency& d) {
+                                        return d.addonId == addonId && d.type == ModPlatform::DependencyType::REQUIRED;
+                                    });
+            dep != deps.end()) {
+            req.append(task->getName());
+        }
+    }
+    return req;
+}
+
 void ResourceDownloadDialog::confirm()
 {
+    auto confirm_dialog = ReviewMessageBox::create(this, tr("Confirm %1 to download").arg(resourcesString()));
+    confirm_dialog->retranslateUi(resourcesString());
+
+    if (auto task = getModDependenciesTask(); task) {
+        connect(task.get(), &Task::failed, this,
+                [&](QString reason) { CustomMessageBox::selectable(this, tr("Error"), reason, QMessageBox::Critical)->exec(); });
+
+        connect(task.get(), &Task::succeeded, this, [&]() {
+            QStringList warnings = task->warnings();
+            if (warnings.count()) {
+                CustomMessageBox::selectable(this, tr("Warnings"), warnings.join('\n'), QMessageBox::Warning)->exec();
+            }
+        });
+
+        // Check for updates
+        ProgressDialog progress_dialog(this);
+        progress_dialog.setSkipButton(true, tr("Abort"));
+        progress_dialog.setWindowTitle(tr("Checking for dependencies..."));
+        auto ret = progress_dialog.execWithTask(task.get());
+
+        // If the dialog was skipped / some download error happened
+        if (ret == QDialog::DialogCode::Rejected) {
+            QMetaObject::invokeMethod(this, "reject", Qt::QueuedConnection);
+            return;
+        } else
+            for (auto dep : task->getDependecies()) {
+                addResource(dep->pack, dep->version);
+            }
```

Comment:
```suggestion
        } else {
            for (auto dep : task->getDependecies())
                addResource(dep->pack, dep->version);
        }
```

less confusing imo

or
```suggestion
        } else
            for (auto dep : task->getDependecies())
                addResource(dep->pack, dep->version);
```

but it feels weird with else tbh